### DISCUSSION TITLE
TShock fails to run, "You must install .NET to run this application." even if Mono JIT compiler version 6.8.0.105 is successfully installed.

### DISCUSSION THREAD

**Discussion by creeloper27** [Pryaxis/TShock#2930] [Q&A]
* TShock version: 5.1.3 (latest at the time of writing)

#### Reproduction steps

1. install Debian 11 (Bullseye) on a x64 system
2. run `sudo apt install mono-complete`
3. cd to a dedicated folder (in my case: `/home/user/servers-and-services/terraria-server/`)
4. get the latest TShock version ([TShock-5.1.3-for-Terraria-1.4.4.9-linux-x64-Release.zip](https://github.com/Pryaxis/TShock/releases/download/v5.1.3/TShock-5.1.3-for-Terraria-1.4.4.9-linux-x64-Release.zip)) and unzip it (I used `wget https://github.com/Pryaxis/TShock/releases/download/v5.1.3/TShock-5.1.3-for-Terraria-1.4.4.9-linux-x64-Release.zip` and `7z x TShock*`)
5. run `chmod u+x TShock.Server`
6. run `./TShock.Server`
7. TShock fails to run

#### running `sudo apt install mono-complete` successfully:
```
>sudo apt install mono-complete
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following NEW packages will be installed:
mono-complete
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 36.9 kB of archives.
After this operation, 77.8 kB of additional disk space will be used.
Get:1 http://deb.debian.org/debian bullseye/main amd64 mono-complete amd64 6.8.0.105+dfsg-3.2 [36.9 kB]
Fetched 36.9 kB in 0s (416 kB/s)
Selecting previously unselected package mono-complete. 
(Reading database ... 368484 files and directories currently installed.)
Preparing to unpack .../mono-complete_6.8.0.105+dfsg-3.2_amd64.deb ...
Unpacking mono-complete (6.8.0.105+dfsg-3.2) ...
Setting up mono-complete (6.8.0.105+dfsg-3.2) ...
Scanning processes...
Scanning processor microcode...
Scanning linux images...
 
Running kernel seems to be up-to-date.
 
The processor microcode seems to be up-to-date.
 
No services need to be restarted.
 
No containers need to be restarted.
 
No user sessions are running outdated binaries.
```

#### running `./TShock.Server` unsuccessfully:
```
>./TShock.Server
You must install .NET to run this application.
 
App: /home/user/servers-and-services/terraria-server/TShock.Server 
Architecture: x64
App host version: 6.0.11 
.NET location: Not found 
 
Learn about runtime installation:
https://aka.ms/dotnet/app-launch-failed
 
Download the .NET runtime: 
https://aka.ms/dotnet-core-applaunch?missing_runtime=true&arch=x64&rid=debian.11-x64&apphost_version=6.0.11
```

#### other notes:
- I have tried using mono on other applications and it seems to work, here is also the output of `mono --version`
```
>mono --version               
Mono JIT compiler version 6.8.0.105 (Debian 6.8.0.105+dfsg-3.2 Tue Jun 29 21:01:01 UTC 2021)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       altstack
        Notifications: epoll
        Architecture:  amd64
        Disabled:      none
        Misc:          softdebug 
        Interpreter:   yes
        LLVM:          supported, not enabled.
        Suspend:       hybrid
        GC:            sgen (concurrent by default)
```

- I also tried to install mono using the tutorial in the docs [here](https://tshock.readme.io/v4.3.22/docs/getting-started#:~:text=%F0%9F%9A%A7-,Installing%20Mono%3F,-TShock%20requires%20the) but unfortunately there is no mono repository for Debian Bullseye (11) at the moment

**Comment by pcider** [comment]
Mono isn't required at all. You would need to install [the .NET6 runtime](https://learn.microsoft.com/en-us/dotnet/core/install/linux-debian)

**Reply by creeloper27** [reply]
I'm confused, if this is the case, why is it told in the official installation guide to use Mono instead? (I'm referring to [this](https://tshock.readme.io/v4.3.22/docs/getting-started#:~:text=%F0%9F%9A%A7-,Installing%20Mono%3F,-TShock%20requires%20the))

**Reply by creeloper27** [reply]
Also, if this again is the case, why is .NET6 required? The vanilla terraria server as of 1.4.4.9 seems to run fine with Mono.

**Reply by punchready** [reply]
The site you linked has a blood red notice on top:
> Hey! These docs are for version 4.3.22, which is no longer officially supported. Click here for the latest version, 4.4.0!

Clicking this note leads you to the mentioned later version, which states:
> These docs are old!
> We've migrated our documentation to another location!
> Please read the [new docs](https://ikebukuro.tshock.co/)!

And those linked new docs correctly inform about .NET 6.
The .NET 6 runtime is not technically necessary, but it is the proper way of running .NET applications as it now has native support for both Windows and Unix on various CPU architectures, which makes Mono more or less obsolete (to my knowledge).

**Reply by sgkoishi** [reply]
> Also, if this again is the case, why is .NET6 required? The vanilla terraria server as of 1.4.4.9 seems to run fine with Mono.

.NET also has higher performance - for some compute-intensive tasks, it can be 10x faster than Mono. tModLoader is also using .NET (6 or 7 I guess). It is a new, officially supported, successor of the old .Net Framework and Mono, just like updating legacy dependencies.

**Reply by creeloper27** [reply]
Thanks, I did not notice that notice on top of the old docs, that's totally on me, my bad 😅
